### PR TITLE
feat(text): add new text component

### PR DESCRIPTION
**Pull request by cnairi**
## DESCRIPTION

Added new text component with css module.
- Tests updated
- Styles updated
- Doc updated

## HOW TO TEST

yarn start
http://localhost:3020/components/text

## SCREENSHOTS / SCREEN RECORDINGS

https://github.com/user-attachments/assets/72f6ac1c-a86d-40ce-83ef-0546cc572999

## COMPATIBILITY

- [x] Tested on Safari (desktop)
- [x] Tested on Chrome (desktop)
- [x] Tested on Firefox (desktop)
- [x] Tested on mobile device sizes
- [x] Tested on tablet device sizes
- [ ] Tested on IOS Safari (either device or simulator)

## QA

- [x] Thoroughly tested in local environment
- [x] Added tests for all new features
- [ ] Added tests that considered edge cases

### CODE REVIEW

File: lib/src/components/Text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        ref,
+        style: { ...cssVariables } as CSSProperties,
+        ...rest,
+      },
+      children
+    )
+  }
+)
+
+Text.displayName = 'Text'
```
Comment (theo-mesnil) [inline lib/src/components/Text/index.tsx]: This is mandatory ?

File: lib/src/components/Text/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    )
+
+    const cssVariables = isMultiLine || isSingleLine ? { '--lineClamp': lines } : {}
+
+    return createElement(
+      tagName,
+      {
+        className: classes,
+        ref,
+        style: { ...cssVariables } as CSSProperties,
```
Comment (theo-mesnil) [inline lib/src/components/Text/index.tsx:62]: Can we avoid style ?

Comment (cnairi) [reply lib/src/components/Text/index.tsx:62]: I really don't have a better approach for now, question asked on our channel
